web/src/beta/lib/core/engines/Cesium/useOverrideGlobeShader.ts

```diff
@@ -218,12 +218,15 @@ export const useOverrideGlobeShader = ({
     globe.vertexShadowDarkness = globeShadowDarkness ?? globe.vertexShadowDarkness;
   }, [cesium, globeShadowDarkness, hasVertexNormals]);
 
-  // This need to be invoked before Globe is updated.
-  useEffect(() => {
+  const originalGlobeFS = useRef<string>();
```

Comment:
What is the point of this? You don't use it unless you are setting `GlobeFS`. But `GlobeFS`'s value comes either from this `originalGlobeFS` or `baseFragmentShaderSource....` (in that order). 
My understanding of this code is:
First render
1. `GlobeFS` is set to `baseFragmentSha.....`
2. `originalGlobeFS` is set to same
Second + render
1. `GlobeFS` is again set to originalGlobeFS
2. `originalGlobeFS` has a value, so skips.

This means that GlobeFS is ALWAYS going to be originalGlobeFS. I'm guessing you want GlobeFS to be `baseFragmentSh....`'s up-to-date value, no? 🤔

Reply:
Fixed! https://github.com/reearth/reearth/pull/951/commits/20ede6c3718a44e2a174a2d37ca1cacaaf572a88